students/BrianNisonger/Session8/circle.py

```diff
@@ -0,0 +1,85 @@
+import math
+
+
+class circle():
+    def __init__(self, radius):
+        self._radius = radius
+
+    @property
+    def radius(self):
+        return self._radius
+
+    @radius.setter
+    def radius(self, radius):
+        self._radius = radius
+
+    @property
+    def diameter(self):
+        return self._radius * 2
+
+    @diameter.setter
+    def diameter(self, value):
+        self._radius = value / 2
+
+    @property
+    def area(self):
+        return math.pi * (self._radius**2)
+
+    @area.setter
+    def area(self, value):
+        self._radius = math.sqrt(value / math.pi)
+
+    @classmethod
+    def from_diameter(circle_object, value):
```

Comment:
we usually use a generic name here, like "cls" or "klass" (can't use "class", it's reserved).

while this will be the Circle class in this case, if you subclass it, it's not longer a Circle -- it will be whatever the subclass is -- a Sphere, for instance.